src/devices/Axp192/M5ToughPowerControl.cs

```diff
@@ -0,0 +1,228 @@
+            // Pull GPIO1 low for resetting the touch driver IC as well
+            _axp.WriteGpioValue(1, PinValue.Low);
+            Thread.Sleep(100);
+            _axp.WriteGpioValue(4, PinValue.High);
+            _axp.WriteGpioValue(1, PinValue.High);
+            // Ensure GPIO2 (Speaker enable) is high
+            _axp.WriteGpioValue(2, PinValue.High);
+            Thread.Sleep(100);
+
+            // Configure charging for external battery (not fitted to M5Tough by default)
+            _axp.SetChargingFunctions(true, ChargingVoltage.V4_2, ChargingCurrent.Current450mA, ChargingStopThreshold.Percent10);
+        }
+
+        /// <summary>
+        /// Beeps using the speaker for the given time, using a fixed frequency
+        /// </summary>
+        /// <param name="duration">The duration of the beep</param>
+        public void Beep(TimeSpan duration)
+        {
+            bool isEnabled = EnableSpeaker;
+            EnableSpeaker = true;
+            using var pwm = _board.CreatePwmChannel(0, 2);
+            pwm.DutyCycle = 0.5f;
+            pwm.Start();
+            Thread.Sleep(duration);
+            pwm.Stop();
+            EnableSpeaker = isEnabled;
+        }
+
+        /// <summary>
+        /// Set LCD backlight voltage. Valid values range from 1.8 - 3.3V. Low values switch off the backlight completely
+        /// </summary>
+        /// <param name="voltage">The voltage to set</param>
+        public virtual void SetLcdVoltage(ElectricPotential voltage)
+        {
+            if (voltage.Millivolts < 1800)
+            {
+                _axp.SetLdoEnable(3, false);
+            }
+            else
+            {
+                _axp.SetLdoEnable(3, true);
+            }
+
+            _axp.SetLdoOutput(3, voltage);
+            _logger.LogInformation($"LCD voltage set to {voltage}");
+        }
+
+        /// <summary>
+        /// Enter and leave low-power mode.
+        /// The physical power button that is normally used to recover the AXP from sleep mode is not accessible on the M5Tough when the case is closed.
+        /// After setting the AXP to sleep, we send the CPU to sleep as well. It will only wake up on tapping the screen.
+        /// </summary>
+        /// <param name="enterSleep">True to enter sleep, false to recover from sleep.</param>
+        /// <remarks>After recovering from sleep mode, some peripheral devices might need to be restarted (such as the display controller)</remarks>
+        public void Sleep(bool enterSleep)
+        {
+            _axp.SetSleep(enterSleep, false);
+            if (enterSleep)
+            {
+                SendCpuToSleep();
+            }
+        }
+
+        private void SendCpuToSleep()
+        {
+            // Sends the board to sleep mode
+            TimeSpan sleepDelay = TimeSpan.FromSeconds(10);
+            const int wakeupPin = 39; // The screen interrupt is connected to this pin
+
+            if (!(_board is ArduinoBoard ab))
```

Comment:
why does Axp192 need to depend on arduino board? this complicates porting to things like nano or using this library standalone

Reply:
It's not the AXP that depends on the board, but the M5Though device. This is a device containing an AXP, an ESP32, a display and several other components, and for this library, Arduino is currently the only way to talk to it. In here, we need to send some special commands to the CPU.